src/Bladeburner/Actions/TeamCasualties.ts

```diff
@@ -0,0 +1,49 @@
+import type { getRandomIntInclusive } from "../../utils/helpers/getRandomIntInclusive";
+
+export enum CasualtyFactor {
+  LOW_CASUALTIES = 0.5, // 50%
+  HIGH_CASUALTIES = 1, // 100%
+}
+
+export interface OperationTeam {
+  teamSize: number;
+  teamLost: number;
+
+  killSupportingSleeves(sleeveDeaths: number): void;
+}
+
+/**
+ * Some actions (Operations and Black Operations) use teams for success bonus
+ * and may result in casualties, reducing the player's hp, killing team members
+ * and killing sleeves (to shock them, sleeves are immortal)
+ *
+ * Caveats: TeamSize = Human Team + Supporting Sleeves set to this action
+ */
+export class TeamCasualties {
+  constructor(private severity: CasualtyFactor, private teamSizeUsed: number, private supportingSleeves: number) {}
```

Comment:
I had to go look up parameter properties to understand what the heck was going on here.

I'm not a fan of this syntax. Almost everything in TypeScript does not affect the runtime semantics; you can trivially strip it out and the JS program remains unchanged. This is different; if you remove the "private"s here, it means something different. IMO we should explicitly use regular JS syntax here, for readability.

Reply:
Is this a blocker or a nit?

Imho if we don't want this syntax we should automate the check in the linter. The JS target we are using will convert this to javascript in the form of private, non-enumerable properties. Natural JS syntax would involve using hashed properties to denote them private, which goes against the scoping in typescript (as you can make them public).

Reply:
I checked the codebase and the only one other place constructor promotion is used is
../Casino/CardDeck/Deck.ts
8:  constructor(private numDecks = 1) {

So since this is "rare" in this codebase, I'm happy to drop it in favor of an object parameter. Though we're redesigning this bit anyway by the looks of it.